src/OpenTelemetry/Trace/ActivityExporterSync.cs

```diff
@@ -0,0 +1,75 @@
+﻿// <copyright file="ActivityExporterSync.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Enumeration used to define the result of an export operation.
+    /// </summary>
+    public enum ExportResultSync
+    {
+        /// <summary>
+        /// Batch is successfully exported.
+        /// </summary>
+        Succeeded = 0,
+
+        /// <summary>
+        /// Batch export failed.
+        /// </summary>
+        Failed = 1,
+    }
+
+    /// <summary>
+    /// ActivityExporterSync base class.
+    /// </summary>
+    public abstract class ActivityExporterSync : IDisposable
+    {
+        /// <summary>
+        /// Exports batch of activities.
+        /// </summary>
+        /// <param name="batch">Batch of activities to export.</param>
+        /// <returns>Result of export.</returns>
+        public abstract ExportResult Export(IEnumerable<Activity> batch);
```

Comment:
I think this `IEnumerable` should be replaced with something that doesn't have heap allocation, not trying to cover it in this PR.

Reply:
❤️ this! Just got this merged so we can prevent allocations doing `foreach`s on the Activity data (tags, links, events, etc.): https://github.com/dotnet/runtime/pull/40544

Reply:
#1080 for tracking.